Aplib.Core/Logging/LogNode.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+
+namespace Aplib.Core.Logging
+{
+    /// <summary>
+    /// Represents a node in the log tree.
+    /// </summary>
+    public class LogNode
+    {
+        /// <summary>
+        /// The loggable object of the node.
+        /// </summary>
+        public ILoggable Loggable { get; }
+
+        /// <summary>
+        /// The depth at which this node resides.
+        /// </summary>
+        public int Depth { get; }
+
+        /// <summary>
+        /// The children of the node.
+        /// </summary>
+        public List<LogNode> Children { get; }
+
+        /// <summary>
+        /// Initialize a new <see cref="LogNode" /> from a given loggable object,
+        /// the depth, and optionally a list of children.
+        /// </summary>
+        /// <param name="loggable">The loggable object of the node.</param>
+        /// <param name="depth">The depth at which this node resides.</param>
+        /// <param name="children">The children of the node. If omitted, an empty list will be used.</param>
+        /// <remarks> It is assumed the children have a correct depth set. </remarks>
```

Comment:
Could you move this to the children parameter? This is now inherited by the other constructor (and also makes sense to put it in the parameter description imo)